tools/check_format.py

```diff
@@ -85,16 +86,46 @@ def isBuildFile(file_path):
   return False
 
 
-def checkFileContents(file_path):
-  message = "%s has over-enthusiastic spaces:" % file_path
-  findSubstringAndPrintError('.  ', file_path, message)
+def headerSubdirDict():
+  subdir_dict = {}
+  for subdir in common.includeDirOrder():
+    subdir_dict[subdir] = True
+  return subdir_dict
+
+SUBDIR_DICT = headerSubdirDict()
+INCLUDE_ANGLE = "#include <"
+INCLUDE_ANGLE_LEN = len(INCLUDE_ANGLE)
+
+def hasInvalidAngleBracketDirectory(line):
+  if not line.startswith(INCLUDE_ANGLE):
+    return False
+  path = line[INCLUDE_ANGLE_LEN:]
+  slash = path.find("/")
+  if slash == -1:
+    return False
+  subdir = path[0:slash]
+  return SUBDIR_DICT.has_key(subdir)
 
+def printLineError(path, line_number, message):
+  printError("%s:%d: %s" % (path, line_number, message))
+
+def checkFileContents(file_path):
+  line_number = 0
+  for line in fileinput.input(file_path):
```

Comment:
`for line_number, line in enumerate(fileinput.input(file_path)):`

Reply:
done...annoyingly they are zero-based.